This commit:
   "Optimizations", 26 Feb 2022:
   https://gitlab.com/nick87720z/tint2/-/commit/e3699ceb8614ca0b73fa543bd1fcd47b24bee67b
broke save_cache(). Nothing gets written to the cache and .xsession-errors shows:
   tint2: Saving icon theme cache...
   tint2: Could not save icon theme cache!

The commit changed 'fopen(cache_path, "w")' to 'fdopen(fd, "w")', trying to
reuse an fd that is already open for the same file (for locking). But that fd
was opened O_RDONLY and fdopen() doesn't allow an incompatible mode (opening
an O_RDONLY fd for writing).

This patch restores the original fopen(cache_path, "w") and the cache works again.

--- tint2-v17.1.3/src/util/cache.c.orig	2022-03-24 05:49:12.000000000 -0400
+++ tint2-v17.1.3/src/util/cache.c	2025-01-21 00:31:18.178933263 -0500
@@ -115,7 +115,7 @@
     }
     flock(fd, LOCK_EX);
 
-    FILE *f = fdopen(fd, "w");
+    FILE *f = fopen(cache_path, "w");
     if (!f) {
         fprintf(stderr, RED "tint2: Could not save icon theme cache!" RESET "\n");
         goto unlock;
